File: neetcode-150/Arrays_and_hashing/Encode_and_decode/solution.py
class Solution:
    def encode(self, strs: List[str]) -> str:
        res = ""
        for s in strs:
            res += str(len(s)) + "#" + s
        return res

    def decode(self, s: str) -> List[str]:
        i, res = 0, []

        while i < len(s):
            j = i
            
            while s[j] != "#":
                j += 1
            length = int(s[i:j])
            j += 1
            res.append(s[j: j + length])
            i = j + length

        return res

File: neetcode-150/Arrays_and_hashing/Encode_and_decode/solution.js
class Solution {
    
    encode(strs) {
        let res = ""

        for(let i = 0; i < strs.length; i++){
            res += strs[i].length + "#" + strs[i];
        }
        return res;
    }
        
    
    decode(str) {
        let i = 0, res = [];

        while(i < str.length){
            let j = i;
            while(str[j] !== "#"){
                j += 1
            }
            const length = Number(str.slice(i , j))
            j += 1;
            res.push(str.slice(j, j + length));
            i = j + length;
        }

        return res;

    }
}
